site/docs/view-spec.md

```diff
@@ -0,0 +1,254 @@
+| Required | object-type    | String identifying the type of object this metadata file is for. ‘View’ for all objects covered in this spec. |
+| Required | location | Location of the view metadata files |
+| Required | current-version-id | Current / latest version of the view. Set to ‘1’ when the view is first created. |
+| Optional | properties | A string to string map of view properties. Contains pre-set properties such as ‘comment’ describing the view, does not contain arbitrary metadata. |
+| Required | versions | An array of structs describing the last few versions of the view. Controlled by the table property: “version.history.num_entries”. See more below. |
+| Required | version-log | An array of structs describing the log of created versions. See more below. |
+| Optional | schemas | A list of schemas and alias-comment maps stored as objects with schema-id. See more below |
+| Optional | current-schema-id | ID of the current schema of the view |
+
+
+‘schemas’ is an array of structs with a field as shown below:
+
+| Required/Optional | Field Name | Description |
+|-------------------|------------|-------------|
+| Optional | schema | An array of structs describing the schema of the view. This field is the same as the ‘schema’ field from Iceberg table spec. See more below. |
+
+
+‘Versions’ is an array of structs with fields as shown below:
+
+| Required/Optional | Field Name | Description |
+|-------------------|------------|-------------|
+| Required | version-id | Monotonically increasing id indicating the version of the view. Starts with “1”. |
+| Required | timestamp-ms | Timestamp expressed in ms since epoch at which the version of the view was created. |
+| Required | summary | A string to string map of view properties to track version metadata. This field can be used by engines to store any necessary properties. Two currently required properties are described below. |
+| Required | sql | A string representing SQL definition of the view as input |
+| Required | dialect | A string specifying the dialect of the ‘sql’ field above. Used by engines to perform necessary translations to the SQL dialect supported by the engine. |
+| Required | default-catalog | A string that specifies the catalog of the user session when the view was created / replaced. Used to resolve the tables used in the view definition. |
+| Required | default-namespace | An array of strings indicating namespace at the time view was created / replaced. Used similar to ‘default-catalog’ above. |
+| Optional | field-aliases | A list of strings of field aliases E.g. a list of alias_name info specified in the following create view statement. `CREATE VIEW v (alias_name COMMENT 'docs', alias_name2, ...) AS SELECT ...` |
```

Comment:
I am not sure I understand the purpose of this field. Could you please clarify? eg how would this alias be different from `aalias` in `select a as aalias from foo`

Reply:
Aliasing the columns in select star sql. This is supported in Spark, Hive, and a few other engines.

Reply:
sorry @jzhge I am not sure I understand your response.

Reply:
Here is the syntax:
```
CREATE VIEW v1  (alias1 COMMENT 'comment for col1', alias2) AS
SELECT *
FROM tbl
```
Table `tbl` has 2 columns: col1 and col2.

Reply:
You are not the first one who find this feature hard to grasp. Presto does not support it. I'd suggest removing this field for this PR and have a follow-up discussion about the view column alias feature.

Reply:
ah I get it now. Interesting. Yeah, I would prefer to defer this as it sounds rather engine specific.

Reply:
This syntax is actually quite common. It's reasonable to not include it right now to get this in, but I think we'll want to add it fairly soon so that we can store the whole set of `CREATE VIEW` metadata.